lib/Service/FontService.php

```diff
@@ -0,0 +1,243 @@
+		return [
+			'kind' => 'fontconfiguration',
+			'server' => $instanceName . ' (' . $this->url->getBaseUrl() . ')',
+			'fonts' => $list,
+		];
+	}
+
+	/**
+	 * @param string $fileName
+	 * @param $newFileResource
+	 * @return array
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function uploadFontFile(string $fileName, $newFileResource): array {
+		$fontDir = $this->getFontAppDataDir();
+		$newFile = $fontDir->newFile($fileName, $newFileResource);
+		$this->generateFontOverview($newFile);
+		$this->cache->remove('fontFileNames');
+		return [
+			'size' => $newFile->getSize(),
+		];
+	}
+
+	/**
+	 * @param string $fileName
+	 * @return ISimpleFile
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getFontFile(string $fileName): ISimpleFile {
+		$fontDir = $this->getFontAppDataDir();
+		return $fontDir->getFile($fileName);
+	}
+
+	/**
+	 * @param string $fileName
+	 * @return string
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function getFontFileOverview(string $fileName): string {
+		$fontDir = $this->getFontOverviewAppDataDir();
+		return $fontDir->getFile($fileName . '.png')->getContent();
+	}
+
+	/**
+	 * @param string $fileName
+	 * @return void
+	 * @throws NotFoundException
+	 * @throws \OCP\Files\NotPermittedException
+	 */
+	public function deleteFontFile(string $fileName): void {
+		$fontDir = $this->getFontAppDataDir();
+		if ($fontDir->fileExists($fileName)) {
+			$fontDir->getFile($fileName)->delete();
+		}
+
+		$overviewDir = $this->getFontOverviewAppDataDir();
+		if ($overviewDir->fileExists($fileName . '.png')) {
+			$overviewDir->getFile($fileName . '.png')->delete();
+		}
+
+		$this->cache->remove('fontFileNames');
+	}
+
+	/**
+	 * @param ISimpleFile $fontFile
+	 * @return void
+	 */
+	private function generateFontOverview(ISimpleFile $fontFile): void {
+		try {
+			$color = [0, 0, 0];
+			$text = 'Lorem ipsum';
+
+			// we need a temp file because imagettftext can't read the font file from a resource
+			// but just a file path
+			$tmpFontFile = tmpfile();
+			$tmpFontFilePath = stream_get_meta_data($tmpFontFile)['uri'];
+			fwrite($tmpFontFile, $fontFile->getContent());
+			fflush($tmpFontFile);
+
+			$im = imagecreatetruecolor(250, 30);
+			$bg_color = imagecolorallocate($im, 255, 255, 255);
+			$font_color = imagecolorallocate($im, $color[0], $color[1], $color[2]);
+			imagefilledrectangle($im, 0, 0, 399, 29, $bg_color);
+			$ttfResult = imagettftext($im, 20, 0, 0, 22, $font_color, $tmpFontFilePath, $text);
+			// this happens with invalid ttf fonts
+			if ($ttfResult === false) {
+				return;
+			}
+
+			$overviewDir = $this->getFontOverviewAppDataDir();
+			$imageFileResource = $overviewDir->newFile($fontFile->getName() . '.png')->write();
+			if (is_resource($imageFileResource)) {
+				imagepng($im, $imageFileResource);
+				imagedestroy($im);
+			}
+		} catch (\Exception | \Throwable $e) {
```

Comment:
Could add a comment here as to why the empty catch block, and what the behaviour is if we dont generate the overview image?

Reply:
Done